pkg/database/mysql.go

```diff
@@ -0,0 +1,26 @@
+package database
+
+
+type MySQLDatabase struct {
+	DB *bun.DB
+}
+
+var (
+	once sync.Once
+	dbInstance *MySQLDatabase
+)
+
+func NewMySQLDatabase(conf *entities.Config) *MySQLDatabase {
+	once.Do(func() {
+		sqlDB, err := sql.Open("mysql", "")
+		if err != nil {
+			panic(err)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/aakashgusain1990/productivite/pull/8#issuecomment-2416672002>#7006ea</a><span> - 10/16/2024, 06:02 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function panics on database connection error, which is not a recommended practice for production code. It should handle the error gracefully and return it to the caller.
</div>
<div id="fix">
<b>Fix:</b> Modify the function to return an error instead of panicking. Use sync.Once to ensure the database is initialized only once, and return any initialization errors. This allows for more flexible error handling at higher levels of the application.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -13,11 +13,14 @@
 -func NewMySQLDatabase(conf *entities.Config) *MySQLDatabase {
 +func NewMySQLDatabase(conf *entities.Config) (*MySQLDatabase, error) {
 +	var err error
  once.Do(func() {
   sqlDB, err := sql.Open("mysql", "")
   if err != nil {
 -		panic(err)
 +		err = fmt.Errorf("failed to open database connection: %w", err)
 +		return
   }
   dbInstance = &MySQLDatabase{DB: bun.NewDB(sqlDB, mysqldialect.New())}
  })
 -return dbInstance
 +if err != nil {
 +	return nil, err
 +}
 +return dbInstance, nil
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged